crates/cranelift/src/wasmfx/optimized.rs

```diff
@@ -419,6 +417,49 @@ pub(crate) mod typed_continuation_helpers {
             let offset = wasmtime_continuations::offsets::vm_cont_ref::PARENT_CHAIN as i32;
             new_stack_chain.store(env, builder, self.address, offset)
         }
+
+        /// Gets the revision counter the a given continuation
+        /// reference.
+        pub fn get_revision<'a>(
+            &mut self,
+            _env: &mut crate::func_environ::FuncEnvironment<'a>,
+            builder: &mut FunctionBuilder,
+        ) -> ir::Value {
+            if cfg!(feature = "unsafe_disable_continuation_linearity_check") {
+                builder.ins().iconst(I64, 0)
+            } else {
+                let mem_flags = ir::MemFlags::trusted();
+                let offset = wasmtime_continuations::offsets::vm_cont_ref::REVISION as i32;
+                let revision = builder.ins().load(I64, mem_flags, self.address, offset);
+                revision
+            }
+        }
+
+        /// Increments the revision counter on the given continuation
+        /// reference.
+        pub fn incr_revision<'a>(
```

Comment:
I see the point in passing in the previously loaded `revision` value as a parameter instead of re-loading it from the continuation. But the comment should reflect that:
```
/// Sets the revision counter on the given reference to `revision` + 1
```

We should also do
```
if cfg!(debug_assertions) {
        let actual_revision = self.get_revision(builder, env);
        emit_debug_assert_eq!(env, builder, revision, actual_revision);
    }
```

Reply:
I will insert this at the start and another after the set to verify the modified value too (it should catch if we were screw up the offset calculation).